### ISSUE TITLE
Force quit when closing window.

### ISSUE THREAD

**Issue by ghost** [yakyak/yakyak#383]
HI.

YakYak force quit when I just close the window after login. If You close the app without logging out and start it again, it is ok if you close the window. Only doing it when you logout/login and close the window.

Also, if you logout/login and quit the app without closing the window and start it again, there is no problem as well.

I have attached a screen shot of the javascript error displayed when it occurs.

Small update : forgot to mention I'm running with Mac OS X El Capitan, 10.11.6.
MacBook Pro early 2008 (2 Gb RAM).

![screen shot 2016-09-21](https://cloud.githubusercontent.com/assets/22344540/18709263/8ad8e91e-7fff-11e6-94a0-5d4c4c8a2793.jpg)

**Comment by randymorris** [comment]
I can confirm this behavior on OSX running master as of this morning.  Log out via the YakYak menu, log back in, then just close the window with the left window button.

**Comment by David-ANT** [comment]
Yes. YakYak runs fine, except that, but it's kind of annoying...

**Comment by arnriu** [comment]
Can you try this commit, if you know how?

**Comment by David-ANT** [comment]
No I'm sorry, I'm not versed into that, it was just to report the bug... Maybe with some help...

**Comment by arnriu** [comment]
Well, it's fine ;) I think it fixes anyway, I was able to reproduce the error, now I can't reproduce it anymore. I'll submit a PR. @randymorris , maybe can you try it?
Thannks for the report @David-ANT .

**Comment by randymorris** [comment]
@arnriu I made this change locally and rebuilt.  The error no longer happens but the application still closes.

When closing the window in a normal situation it does not kill the application, rather the application stays running in the dock.

**Comment by arnriu** [comment]
Oh I see. Will look into it.

**Comment by arnriu** [comment]
Can you try this one then? I'm not 100% sure about why this was here, so, it needs testing.

**Comment by randymorris** [comment]
Still kills the application for me.

**Comment by arnriu** [comment]
Strange, it doesn't quit anymore for me if i click top left red button, no matter what I try (except on login window itself, which is expected).

**Comment by randymorris** [comment]
Strange.  I just started fresh, deleted my dist/ and node_modules/ folders, re-applied the patch, and rebuilt.  Now it appears to be working fine.  Not sure what happened there.

+1 from me for the change but I can't test any other platforms.

**Comment by averissimo** [comment]
lol, this will be fixed in 1.4.0-beta3

to try it out check #451

**Comment by averissimo** [comment]
@David-ANT  it would be awesome if any of you could try to test to make sure it was corrected and has a good behaviour

**Comment by David-ANT** [comment]
Hi !

Very strange... No more Javascript error but still, after the first login, the app just directly quits if I close the window BUT without any error reported now. It's just like if it has crashed without warning. After starting it again, no problem. 

It exhibits the exact same behavior I reported the first time (see above) but without Javascript / Uncaught Exception popup window. Still, it is less annoying like that, at least for a "standard" non dev. user like me, but there is something strange there... Developer challenge ? ;)

I tried it a dozen of times and it is reproducible every time on my old rig (MacBook Pro Early 2008, 2,4 GHz Intel Core 2 Duo, 2 GB 667 MHz DDR2 SDRAM, NVIDIA GeForce 8600M GT 256 MB and OS X 10.11.6). First time I encounter that bug was on 10.11.5...
